### ISSUE TITLE
CI unpin numpy

### ISSUE THREAD

**Issue by MarcoGorelli** [pandas-dev/pandas#50392]
numpy 1.24.0 can't be used with numba. once https://github.com/numba/numba/pull/8620 is merged, and this is fixed in numba, we can unpin numpy in CI

**Comment by lithomas1** [comment]
Probably can close this once numba releases their 1.24.0 support, and this block is deleted.

https://github.com/pandas-dev/pandas/blob/9e8a243c5b77429b662030cc0128a076eea192c8/pandas/util/_test_decorators.py#L74-L79

**Comment by jbrockmendel** [comment]
according to numba's release notes 0.57.0 supports numpy 1.24.  is this closable?

**Comment by MarcoGorelli** [comment]
yeah, once we verify that removing

https://github.com/pandas-dev/pandas/blob/b947e6a63f555ecf48fff2f488bed3dabe7f1e73/pandas/util/_test_decorators.py#L72-L78

keeps CI green

anyone fancy making a PR?

**Comment by PrathumP** [comment]
Yes, numba's latest version 0.57.0 supports Numpy 1.24.0. You can check it out here https://numba.readthedocs.io/en/stable/release-notes.html#version-0-57-0-1-may-2023.

So, Can I raise a PR just deleting that block ?
